src/chunkserver/datastore/chunkserver_chunkfile.cpp

```diff
@@ -417,6 +419,23 @@ CSErrorCode CSChunkFile::Write(SequenceNum sn,
                    << ",chunk sn: " << metaPage_.sn;
         return errorCode;
     }
+
+    if (chunkrate_.get() && cvar_.get()) {
+        *chunkrate_ += length;
+        uint64_t res = *chunkrate_;
+        auto actualSyncChunkLimits = syncChunkLimits_;
+        // if single write size > syncThreshold, for cache friend to
```

Comment:
It is best to change the rate at the entry of  `CSChunkFile::Write`. And update of `syncChunkLimits_`  can be put into a function like `MayUpdateLimits(uint64_t writelen)`